06-defensive.md

```diff
@@ -238,16 +246,23 @@ Its advocates believe it produces better code faster because:
 
 2. Writing tests helps programmers figure out what the function is actually supposed to do.
 
-Here are three test functions for `range_overlap`:
+Below are three test functions for `range_overlap`, but first we need a brief aside to explain some MATLAB behaviour.
+1. Testing for equality between matrices
+	Using `matA == matB` returns a matrix of logical values
+2. Input format required for assert
+	`assert` requires a scalar logical input argument
```

Comment:
This needs a little refinement. First, it renders incorrectly (see the rendered markdown). Second, there is no connection between the two points, and no explicit connection to the functions `isequal` or `isnan`. A little more explanation could be helpful.